include/revng/Support/FunctionIntrospection.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <functional>
+#include <tuple>
+
+#include "revng/Support/TupleIntrospection.h"
+
+template<class T>
+struct function_traits : function_traits<decltype(&T::operator())> {};
+
+// partial specialization for function type
+template<class R, class... Args>
+struct function_traits<R(Args...)> {
+  using result_type = R;
+  using argument_types = std::tuple<Args...>;
+};
+
+// partial specialization for function pointer
+template<class R, class... Args>
+struct function_traits<R (*)(Args...)> {
+  using result_type = R;
+  using argument_types = std::tuple<Args...>;
+};
+
+// partial specialization for std::function
```

Comment:
Start comments with capital letter.
If this is documentation for the function, use doxygen (`///`).